### DISCUSSION TITLE
iOS 18 build error

### DISCUSSION THREAD

**Discussion by bitomule** [pointfreeco/swift-composable-architecture#3324] [General]
### Description

After updating Xcode to Xcode 17 beta 6 I started to have this build error in the withDependencies function for Swift6

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

Build dependencies without errors

### Actual behavior

Build fails at:

```swift
return try await DependencyValues.$_current.withValue(dependencies) {
```

With error:

````
Returning a task-isolated '@noescape @async @callee_guaranteed @substituted <τ_0_0> () -> (@out τ_0_0, @error any Error) for <R>' value as a 'sending' result risks causing data races
````

### Steps to reproduce

Build using Xcode 16 beta 6

### The Composable Architecture version information

main

### Destination operating system

iOS 16 

### Xcode version information

Version 16.0 beta 6 (16A5230g)

### Swift Compiler version information

```shell
swiftlang-6.0.0.9.10 clang-1600.0.26.2
```

**Comment by stephencelis** [comment]
@bitomule I believe that issue was fixed in a recent version of swift-dependencies. Can you update your packages and try again? And if you're still seeing the issue, please share a project that reproduces the issue.

**Comment by eunryu93** [comment]
@stephencelis 
Hi, I appreciate your answer. 
but, This project is company App Project, so, I cannot share this project..;-;
Instead, I try to make new project, and will check this problem. And if show this error message to new project, I will share project.

And, I have another question related to this issue.
1] Is it possible for TCA errors to differ depending on the Xcode version? I suspect that the issue might not be with the TCA library but with the Xcode compiler.
2] I have a code obfuscation plugin installed in Xcode; could this be affecting it?

I would appreciate your response. Thank you!

**Comment by bitomule** [comment]
> @bitomule I believe that issue was fixed in a recent version of swift-dependencies. Can you update your packages and try again? And if you're still seeing the issue, please share a project that reproduces the issue.

My bad. I can confirm that forcing the update of swift-dependencies fixed the issue. I tried updating main package but that didn't work, I had to force Xcode to update swift-dependencies.

@eunryu93 if you have the same issue try updating package.

thanks @stephencelis

**Comment by andreyz** [comment]
@stephencelis could the issue be because TCA is specifying it's dependency as
`.package(url: "https://github.com/pointfreeco/swift-dependencies", from: "1.3.5"),`? While the Xcode 16 beta 6 fix was shipped in `swift-dependencies` `1.3.9`?

In my complex dependency-wise project, even when I pin TCA to `1.14.0` I can't force `https://github.com/pointfreeco/swift-dependencies` to resolve to `1.3.9` without explicitly including it in dependencies for the app's package target, which shouldn't be required, as I only use `swift-dependencies` transitively via TCA

**Comment by mbrandonw** [comment]
> In my complex dependency-wise project, even when I pin TCA to `1.14.0` I can't force `https://github.com/pointfreeco/swift-dependencies` to resolve to `1.3.9` without explicitly including it in dependencies for the app's package target, which shouldn't be required, as I only use `swift-dependencies` transitively via TCA

This is the intended way to use dependencies. If you are getting a library transitively but you want to make sure it is at a specific version (or higher), then you can add an explicit dependence on the library to set the version you need. It is not expected that the base library is going to supply the exact version you need because different people may want different versions. For example, while 1.3.9 may have fixed an issue for you in the beta, it may have caused a problem for someone in Xcode 15.4. And they will have the freedom to stay at 1.3.5, or whatever version they need. 

In fact, we strive to keep our dependencies as _loose_ as possible to give people maximum flexibility with what versions they need. The only time we bump versions in our dependencies is when there is a fix to a serious problem that everyone should have. But a build error in an Xcode beta does not meet that bar right now.

**Reply by andreyz** [reply]
Thank you for an extensive explanation. This makes sense

**Comment by mbrandonw** [comment]
Since this isn't an issue with the library I am going to convert this to a discussion. Feel free to keep the conversation going over there.
